README.md

```diff
@@ -43,114 +43,280 @@
+]
+```
+
+## Usage
+
+### Setting up Featurevisor SDK
+We would like to be able to set up the Featurevisor SDK instance once and reuse the same instance everywhere.
+
+The SDK can be initialized in two different ways depending on your needs.
+
+### Synchronous
+You can fetch the data file content on your own and just pass it via options.
+
+```swift
+import FeaturevisorSDK
+
+let datafileContent: DatafileContent = ...
+var options: InstanceOptions = .default
+options.datafile = datafileContent
+
+let featurevisor = try createInstance(options: options)
+```
+
+### Asynchronous
+If you want to delegate the responsibility of fetching the datafile to the SDK.
 
 ```swift
-// swift-tools-version:4.0
-import PackageDescription
+import FeaturevisorSDK
 
-let package = Package(
-  name: "ExampleApp",
-  dependencies: [
-    // add the repo with desired semver as a dependency here
-    .package(url: "git@github.com:featurevisor/featurevisor-swift.git", .exact("X.Y.Z"))
-  ],
-  .target(
-    name: "ExampleApp",
-    dependencies: [
-      // add the name of module from the dependency in your desired targets
-      "FeaturevisorSDK"
-    ]
-  )
-)
+var options: InstanceOptions = .default
+options.datafileUrl = "https://cdn.yoursite.com/production/datafile-tag-all.json" 
+let featurevisor = try createInstance(options: options)
 ```
 
-Building your application will fetch and install everything locally:
+If you need to take further control on how the datafile is fetched, you can pass a custom `handleDatafileFetch` function
 
+```swift
+public typealias DatafileFetchHandler = (_ datafileUrl: String) -> Result<DatafileContent, Error>
 ```
-$ swift build
+
+```swift
+import FeaturevisorSDK
+
+var options: InstanceOptions = .default
+options.handleDatafileFetch = { datafileUrl in
+    ...
+}
+let featurevisor = try createInstance(options: options)
 ```
 
-## Usage
+### Context
+Contexts are a set of attribute values that we pass to SDK for evaluating features.
 
-@TODO: ...
+They are objects where keys are the attribute keys, and values are the attribute values.
+
+```swift
+public enum AttributeValue {
+    case string(String)
+    case integer(Int)
+    case double(Double)
+    case boolean(Bool)
+    case date(Date)
+```
+
+```swift
+let context = [
+  "myAttributeKey": .string("myStringAttributeValue"),
+  "anotherAttributeKey": .double(0.999),
+]
+```
+
+### Checking if enabled
+Once the SDK is initialized, you can check if a feature is enabled or not:
+
+```swift
+let featureKey = "my_feature";
+let context = [
+    "userId": .string("123"),
+    "country": .string("nl")
+]
+
+let isEnabled = sdk.isEnabled(featureKey: featureKey, context: context)
+```
+
+### Getting variations
+If your feature has any variations defined, you can get evaluate them as follows
+
+```swift
+let featureKey = "my_feature";
+let context = [
+    "userId": .string("123")
+]
+
+let variation = sdk.getVariation(featureKey: featureKey, context: context)
```

Comment:
idea for future: we may consider using the `_` thingy in Swift, so we don't always have to pass the argument names manually. 

we could then do:

```swift
let variation = sdk.getVariation(featureKey, context)
```

Reply:
I was fighting with my thoughts about this some time ago trying to find the answer to what will be more obvious for people using it. Will take it as an improvement and handle under different PR.